### PR TITLE
Skip GDS notification forwarding if driver configured

### PR DESCRIPTION
**Pull request by FlorentinD**

### CODE REVIEW

**Comment by FlorentinD** [comment]
[Catch new driver warnings](https://trello.com/c/EzU3gNcE/9651-catch-new-driver-warnings)

File: graphdatascience/query_runner/neo4j_query_runner.py [hunk 1]
```diff
@@ -124,10 +133,19 @@ def run_cypher(
...
+            ):
+                notifications = result.consume().notifications
+                if notifications:
+                    for notification in notifications:
+                        self._forward_cypher_warnings(notification)
+            elif Neo4jQueryRunner._NEO4J_DRIVER_VERSION > ServerVersion(5, 21, 0):
+                # the client does not expose YIELD fields so we just skip these warnings for now
+                warnings.filterwarnings(
+                    "ignore", message=r".*The query used a deprecated field from a procedure\. .* by 'gds.* "
+                )
```
Comment (knutwalker) [inline graphdatascience/query_runner/neo4j_query_runner.py:148]: I did think a while about what first this check was doing and eventually realized it is to say that every notification that isn't even a warning in the first place should be forwarded anyways. Maybe it's also the room temperature that it took a while to understand 😅 

I wonder if flipping the conditions would make it a bit easier to read, maybe:


```suggestion
            if (
                Neo4jQueryRunner._NEO4J_DRIVER_VERSION >= ServerVersion(5, 21, 0)
                and result._warn_notification_severity == "WARNING"
            ):
                # the client does not expose YIELD fields so we just skip these warnings for now
                warnings.filterwarnings(
                    "ignore", message=r".*The query used a deprecated field from a procedure\. .* by 'gds.* "
                )
            else:
                notifications = result.consume().notifications
                if notifications:
                    for notification in notifications:
                        self._forward_cypher_warnings(notification)
```

File: graphdatascience/query_runner/neo4j_query_runner.py [hunk 2]
```diff
@@ -124,10 +133,19 @@ def run_cypher(
...
-                    self._forward_cypher_warnings(notification)
+            if (
+                Neo4jQueryRunner._NEO4J_DRIVER_VERSION < ServerVersion(5, 21, 0)
+                or result._warn_notification_severity != "WARNING"
+            ):
+                notifications = result.consume().notifications
+                if notifications:
+                    for notification in notifications:
+                        self._forward_cypher_warnings(notification)
+            elif Neo4jQueryRunner._NEO4J_DRIVER_VERSION > ServerVersion(5, 21, 0):
```
Comment (knutwalker) [inline graphdatascience/query_runner/neo4j_query_runner.py:144]: this should be >=, no?

```suggestion
            elif Neo4jQueryRunner._NEO4J_DRIVER_VERSION >= ServerVersion(5, 21, 0):
```

File: graphdatascience/tests/integration/test_error_handling.py [hunk 3]
```diff
@@ -189,3 +193,30 @@ def test_auto_completion_false_positives(gds: GraphDataScience) -> None:
...
+
+
+def test_forward_server_side_warning(gds: GraphDataScience) -> None:
+    with pytest.raises(Warning, match="The query used a deprecated function: `id`."):
+        gds.run_cypher("MATCH (n) RETURN id(n)")
+
+
+@pytest.mark.filterwarnings("ignore: notification warnings are a preview feature")
+def test_forward_driver_configured_warning() -> None:
+    if Neo4jQueryRunner._NEO4J_DRIVER_VERSION > ServerVersion(5, 21, 0):
```
Comment (knutwalker) [inline graphdatascience/tests/integration/test_error_handling.py:205]:
```suggestion
    if Neo4jQueryRunner._NEO4J_DRIVER_VERSION >= ServerVersion(5, 21, 0):
```

File: graphdatascience/tests/integration/test_error_handling.py [hunk 4]
```diff
@@ -189,3 +193,30 @@ def test_auto_completion_false_positives(gds: GraphDataScience) -> None:
...
+            gds.run_cypher("MATCH (n) RETURN id(n)")
+
+
+def test_filter_out_client_related__warning(gds: GraphDataScience) -> None:
+    gds.graph.drop("g", failIfMissing=False)
+
+
+@pytest.mark.filterwarnings("ignore: notification warnings are a preview feature")
+def test_filter_out_client_related__driver_configured_warning() -> None:
+    if Neo4jQueryRunner._NEO4J_DRIVER_VERSION > ServerVersion(5, 21, 0):
```
Comment (knutwalker) [inline graphdatascience/tests/integration/test_error_handling.py:219]:
```suggestion
    if Neo4jQueryRunner._NEO4J_DRIVER_VERSION >= ServerVersion(5, 21, 0):
```

**Review by knutwalker — APPROVED** [review]
🤞

File: graphdatascience/query_runner/neo4j_query_runner.py (hunk 1 above)
Comment (FlorentinD) [reply graphdatascience/query_runner/neo4j_query_runner.py]: I also confused myself while writing. Thanks for the idea :)

**Pull request merged**